src/System.Private.ServiceModel/tests/Stress/SharedPoolsOfWCFObjects/SharedPoolsOfWCFObjects.cs

```diff
@@ -189,33 +314,47 @@ public static void RecyclePooledFactories()
         }
     }
 
-    public class RecyclablePooledFactoriesAndChannels_OpenOnce
+    public static class RecyclablePooledFactoriesAndChannels_OpenOnce<ChannelType, TestTemplate>
+        where TestTemplate : ITestTemplate<ChannelType>, new()
     {
         private static int s_iteration = 0;
-        private static CallStats s_CreateChannelStats = new CallStats(samples: 1000, errorSamples: 1000);
-        private static CallStats s_CallChannelStats = new CallStats(samples: 1000000, errorSamples: 10000);
-        private static CallStats s_CloseChannelStats = new CallStats(samples: 1000, errorSamples: 1000);
-        private static CallStats s_CloseFactoryStats = new CallStats(samples: 1000, errorSamples: 1000);
+        private static CallStats s_CreateChannelStats;
+        private static CallStats s_CallChannelStats;
+        private static CallStats s_CloseChannelStats;
+        private static CallStats s_CloseFactoryStats;
+
+        private static ITestTemplate<ChannelType> s_test;
+
         private static PoolOfThings<FactoryAndPoolOfItsObjects<
-                ChannelFactory<WcfService1.IService1>,
-                OpenOnceChannelWrapper<WcfService1.IService1>>
+                ChannelFactory<ChannelType>,
+                OpenOnceChannelWrapper<ChannelType>>
             > s_recyclablePooledFactoriesAndChannels;
 
         static RecyclablePooledFactoriesAndChannels_OpenOnce()
         {
+            // moving the following magic numbers to the test template interface will help each test to collect different type of stats
+            s_CreateChannelStats = new CallStats(samples: 1000, errorSamples: 1000);
+            s_CallChannelStats = new CallStats(samples: 1000000, errorSamples: 10000);
```

Comment:
For the future -- more magic #'s that should become TestProperties or const's.  Not required for merge.